doc/designs/edns.md

```diff
@@ -0,0 +1,198 @@
+
+## Goals
+
+The primary goal is to provide a way to deploy integrated DNS server with the enforcement of using DNS over encrypted channels instead of using standard UDP port 53 with unencrypted communication. This means that DNS clients must talk to DNS servers using DoT method that is fully encrypted. The scope for the traffic encryption is for all DNS queries that are leaving the host, while the local communication within the host does not require encryption.
+
+
+## Design Choices
+
+This feature builds upon the existing DNS component in FreeIPA, maintaining its core assumptions and goals while introducing secure communication through DoT. It is designed to enhance FreeIPA's DNS deployment options and address the growing need for encrypted DNS communication. This first version of the feature is limited and restricted as we are aiming for a simple implementation that covers basic use cases. These restrictions will be addressed in next iterations of the feature development.
+
+This initial version of DoT is supported only for new installations/deployments, although administrators can run `ipa-dns-install` with the new DoT options to redeploy the integrated DNS service with DoT enabled on an existing deployment.
+
+The deployment of the DNS service will depend on the infrastructure configuration. During server/replica/client deployment or client/replica enrollment, both unencrypted and encrypted communications will be allowed as default. However, for more restrictive scenarios where only encrypted communications are allowed, a new option is needed to enforce this behavior. The new option `--dns-policy` will have two choices: `relaxed` will allow unencrypted DNS queries, while `enforced` will restrict unencrypted queries on port 53.
+
+The implementation of DoT in FreeIPA's DNS service leverages the following design choices:
+
+- Current FreeIPA works with `Bind` 9.18. This version of BIND doesn't support DoT-forwarding. Therefore, for server/replica deployment, we rely on an additional component, `unbound`, which redirects encrypted DNS queries to the external DNS server. This includes enabling TLS connections on TCP port 853 and managing TLS certificates for both components `Bind` and `Unbound`. The upgrade to the latest LTS `Bind` version (9.20) is already in progress but not completed. Once this upgrade is complete, the design will be simplified by removing `unbound` as Bind will fully support DoT-forwarding. 
+
+- Configuration Management: New LDAP attributes and FreeIPA CLI options are introduced to enable and configure DoT settings.
+
+- Certificate Integration: FreeIPA's certificate management framework can automatically be used to generate and manage TLS certificates giving the administrators the choice to either provide a certificate or rely on custodia for certificate auto enrollment if no certificate is provided. These options are handled by the proper CLI options (please check section [CLI](#cli)).
```

Comment:
Just for clarification, custodia is doing certificate auto enrollment?

Reply:
When IPA CA is used, yes, that's right.